src/hypothesis/strategies.py

```diff
@@ -945,11 +955,26 @@ def builds(target, *args, **kwargs):
     the target.
 
     """
+    if target_and_args:
+        target, args = target_and_args[0], target_and_args[1:]
+        if not callable(target):
+            raise InvalidArgument('The first positional argument to builds() must be a callable'
+                                  'target to construct.')
+    elif 'target' in kwargs:
+        args = []
+        note_deprecation('Specifying the target as a keyword argument to builds() is deprecated. '
+                         'Provide it as the first positional argument instead.')
+        target = kwargs.pop('target')
+    else:
+        raise InvalidArgument('No target was provided to builds().'
+                              'Specify it as the first positional argument.')
+
     if infer in args:
         # Avoid an implementation nightmare juggling tuples and worse things
         raise InvalidArgument('infer was passed as a positional argument to '
                               'builds(), but is only allowed as a keyword arg')
-    hints = get_type_hints(target.__init__ if isclass(target) else target)
+    hints = get_type_hints(target.__init__
```

Comment:
No need to change this line.

Reply:
oh right, sorry, that was for line length in the older version that introduced the longer name `hypothesis_internal_target`